File: app/src/main/java/com/example/android/todolist/addtaskactivity/AddTaskViewModelFactory.java
package com.example.android.todolist.addtaskactivity;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.example.android.todolist.database.AppDatabase;

/**
 * Created by raed on 9/20/18.
 */

public class AddTaskViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private final AppDatabase database;
    private final int taskId;

    public AddTaskViewModelFactory (AppDatabase database, int taskId) {
        this.database = database;
        this.taskId = taskId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new AddTaskViewModel(database, taskId);
    }
}

File: app/src/main/java/com/example/android/todolist/addtaskactivity/AddTaskViewModelFactory.kt
package com.example.android.todolist.addtaskactivity

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.example.android.todolist.database.AppDatabase

/**
 * Created by Raed Saeed on 8/3/2019.
 */
class AddTaskViewModelFactory(private val appDatabase: AppDatabase, private val taskId : Int) : ViewModelProvider.NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return AddTaskViewModel(appDatabase, taskId) as T
    }
}